modules/experimental/src/gltf/gltf-material-parser.js

```diff
@@ -140,4 +145,61 @@ export default class GLTFMaterialParser {
       });
     }
   }
+
+  renameShaderVariable(oldName, newValue) {
+    const code = `#define ${oldName} ${newValue}\n`;
+    this.vs = code + this.vs;
+    this.fs = code + this.fs;
+  }
+
+  resolveSemanticUniform(name, {semantic}, value) {
+    // handle texture
+    if (value && value.texture) {
+      if (value.texCoord) {
+        log.warn('Multi tex-coord not supported')();
+      }
+
+      this.parseTexture(value, name);
+    } else if (semantic === 'CESIUM_RTC_MODELVIEW' || semantic === 'MODELVIEW') {
+      this.renameShaderVariable(name, 'u_ModelView');
+    } else if (semantic === 'PROJECTION') {
+      this.renameShaderVariable(name, 'u_Projection');
+    } else {
+      log.warn('Cannot resolve uniform in glTF custom technique')();
+    }
+  }
+
+  parseTechnique(technique) {
+    const {program = {}, attributes = {}, uniforms = {}, values = {}} = technique;
+
+    if (program.vertexShader && program.fragmentShader) {
+      this.vs = program.vertexShader.code;
+      this.fs = program.fragmentShader.code;
+
+      // custom shader will not use defines
+      this.defines = {};
+
+      this.uniforms = {};
+      Object.keys(uniforms).forEach(uniform => {
+        this.resolveSemanticUniform(uniform, uniforms[uniform], values[uniform]);
+      });
+
+      const originalAttributes = this.attributes;
+      this.attributes = {};
+
+      Object.keys(attributes).forEach(attribute => {
```

Comment:
Total nit, prefer for loops on iterators: `for (const attributeName of Object.keys(attributes))`